var/spack/repos/builtin/packages/py-flake8/package.py

```diff
@@ -28,13 +28,40 @@
 class PyFlake8(Package):
     """Flake8 is a wrapper around PyFlakes, pep8 and Ned Batchelder's
     McCabe script."""
-    homepage = "http://flake8.readthedocs.io/en/latest/"
-    url      = "https://pypi.python.org/packages/source/f/flake8/flake8-2.5.4.tar.gz"
 
-    version('2.5.4', 'a4585b3569b95c3f66acb8294a7f06ef')
+    homepage = "https://github.com/PyCQA/flake8"
+    url      = "https://github.com/PyCQA/flake8/archive/3.0.4.tar.gz"
 
-    extends('python')
-    depends_on('py-setuptools', type='build')
+    version('3.0.4', 'cf2a7d8c92070f7b62253404ffb54df7')
+    version('2.5.4', '366dd1de6c300254c830b81e66979f06')
+
+    extends('python', ignore='bin/(pyflakes|pycodestyle)')
+    depends_on('python@2.7:2.8,3.4:')
+
+    # Most Python packages only require py-setuptools as a build dependency.
+    # However, py-flake8 requires py-setuptools during runtime as well.
+    depends_on('py-setuptools', type=nolink)
+
+    # pyflakes >= 0.8.1, != 1.2.0, != 1.2.1, != 1.2.2, < 1.3.0
+    depends_on('py-pyflakes@0.8.1:1.1.0,1.2.3:1.2.3', when='@3.0.4', type=nolink)  # noqa
+    # pyflakes >= 0.8.1, < 1.1
+    depends_on('py-pyflakes@0.8.1:1.0.0', when='@2.5.4', type=nolink)
+
+    # pycodestyle >= 2.0.0, < 2.1.0
+    depends_on('py-pycodestyle@2.0.0:2.0.999', when='@3.0.4', type=nolink)
+    # pep8 >= 1.5.7, != 1.6.0, != 1.6.1, != 1.6.2
+    depends_on('py-pycodestyle@1.5.7,1.7.0:', when='@2.5.4', type=nolink)
+
+    # mccabe >= 0.5.0, < 0.6.0
+    depends_on('py-mccabe@0.5.0:0.5.999', when='@3.0.4', type=nolink)
+    # mccabe >= 0.2.1, < 0.5
+    depends_on('py-mccabe@0.2.1:0.4.0', when='@2.5.4', type=nolink)
```

Comment:
These version constraints are a nightmare. I would say "oh, who cares", but I'm pretty sure setuptools will reject a package if it isn't in that dependency range and download one from PyPi instead. And I would just choose the lowest common denominator and tell them to all depend on the latest version, but they're hard-coded not to allow that. I guess it prevents them from guaranteeing backwards compatibility.